extensions/todoist/src/menubar.tsx

```diff
@@ -9,13 +9,16 @@ import { Task } from "@doist/todoist-api-typescript";
 
 export default function Command() {
   const isTodayView = getPreferenceValues().view === "today";
+  const isUpcomingPlus7View = getPreferenceValues().view === "upcoming_plus_7";
 
   const {
     data: tasks,
     isLoading: isLoadingTasks,
     error: tasksError,
     mutate: mutateTasks,
-  } = useCachedPromise(() => todoist.getTasks({ filter: isTodayView ? "today|overdue" : "all" }));
+  } = useCachedPromise(() =>
+    todoist.getTasks({ filter: isTodayView ? "today|overdue" : isUpcomingPlus7View ? "due before: +7 days" : "all" })
```

Comment:
It's already a preference value.  It's called "Upcoming +7" instead of "Upcoming". Both exists.

However, I could do another preference especially done for Upcoming which could be a number or "all" if you prefer?

Reply:
yep that would make sense

Reply:
Okay I will work in this way. You're right, it's a good point. Cheers!